feature-libs/cart/main/root/components/add-to-cart/add-to-cart.component.ts

```diff
@@ -149,38 +132,22 @@ export class AddToCartComponent implements OnInit, OnDestroy {
     if (!this.productCode || quantity <= 0) {
       return;
     }
-    this.activeCartService
-      .getEntries()
-      .pipe(take(1))
-      .subscribe((entries) => {
-        this.numberOfEntriesBeforeAdd = entries.length;
-        this.openModal();
-        this.activeCartService.addEntry(this.productCode, quantity);
+    this.activeCartService.addEntry(this.productCode, quantity);
```

Comment:
For this component, can we also make cart-lib loaded for login user? Since, we have `mini-cart` and `wish-list` (both make cart-lib loaded for login user), so in PDP or PLP, cart-lib loaded for login user. But, without `mini-cart` or `wish-list`, there will be an issue. That is, when there is a `cartId` stored in `localStorage`, and the cart is removed from backend, the "add-to-cart" will not work well. When the `activeCartService` construct, it will load the cart which `id` is stored in browser or `current`.  If this loading fails, the `cartId` stored in storage will be removed (this is what we expected). But, if "add-to-cart" makes the loading happen (loading `current`, but the cartId is something stored in browser), the state will be wrong.

Reply:
The use case is different than mini-cart or wish list, I'm not sure the same logic applies.  Let's discuss this one.